libs/core/tests/unit_tests/test_tools.py

```diff
@@ -925,6 +930,81 @@ async def foo(bar: str) -> str:
     )
 
 
+@tool
+def foo(bar: Any, bar_config: RunnableConfig) -> Any:
+    """The foo."""
+    assert bar_config["configurable"]["foo"] == "not-bar"
+    assert bar == "baz"
+    return bar
+
+
+@tool
+async def afoo(bar: Any, bar_config: RunnableConfig) -> Any:
+    """The foo."""
+    await asyncio.sleep(0.0001)
```

Comment:
hy sleeping??

Reply:
i am a simple assistant trained by openai. when i see code i copy pasta it. i'll be taking no further questions at this time. because i am incapable of answering them